### PR TITLE
refactor: Remove serialize-javascript and add configOverride to appWithTranslation

### LINKED ISSUE #931: [8.0.0-beta.4] Error when using postProcessors
Hi, first of all, thanks for this plugin, yours and the community's efforts :heart:  

## Describe the bug
I'm unable to use an i18next postprocessor - here [sprintf-postProcessor](https://github.com/i18next/i18next-sprintf-postProcessor)-, similar to [this comment](https://github.com/isaachinman/next-i18next/issues/869#issuecomment-762595861), any `use` config parameter seems to trigger the following error when trying to start the next application (using `next` command):
> SerializableError: Error serializing `._nextI18Next.userConfig.use[0].process` returned from `getStaticProps` in "/".
> Reason: `function` cannot be serialized as JSON. Please only return JSON serializable data types.

## Occurs in next-i18next version
next-18next version: 8.0.0-beta.4 - c3dcef1f2c3510fa6eb2fd5c3fad42ab717964f6

## Steps to reproduce
- Checkout c3dcef1f2c3510fa6eb2fd5c3fad42ab717964f6 in this repository
- Apply "Reproducible patch" bellow
- Run `yarn run-example`

<details>
  <summary>Reproducible patch from c3dcef1f2c3510fa6eb2fd5c3fad42ab717964f6</summary>
  
```diff
diff --git a/examples/simple/next-i18next.config.js b/examples/simple/next-i18next.config.js
new file mode 100644
index 0000000..5366457
--- /dev/null
+++ b/examples/simple/next-i18next.config.js
@@ -0,0 +1,6 @@
+const sprintf = require("i18next-sprintf-postprocessor");
+
+module.exports = {
+  postProcess: "sprintf",
+  use: [sprintf],
+};
diff --git a/examples/simple/package.json b/examples/simple/package.json
index be9e52c..83c434f 100644
--- a/examples/simple/package.json
+++ b/examples/simple/package.json
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "next": "^10.0.0",
+    "i18next-sprintf-postprocessor": "latest",
     "next-i18next": "link:../../"
   },
   "devDependencies": {
```
  
</details>

## Expected behaviour
There should be no errors and I should be able to translate using any backend/plugin

### PR DESCRIPTION
**Pull request by isaachinman**
Fixes #931.

### CODE REVIEW

**Comment by isaachinman** [comment]
@LeonardDrs Do you mind a quick code/docs review?

File: README.md [hunk 1]
```diff
@@ -154,6 +157,28 @@ module.exports = {
...
 ```
 
+#### Unserialisable configs
+
+Some `i18next` plugins (which you can pass into `config.use`) are unserialisable, as they contain functions and other JavaScript primitives.
+
+You may run into this if your use case is more advanced. You'll see NextJs throw an error like:
+
+```
+Error: Error serializing `._nextI18Next.userConfig.use[0].process` returned from `getStaticProps` in "/pre-generated".
```
Comment (LeonardDrs) [inline README.md:167]: `pre-generated` is the name of my next route, maybe we should use a placeholder?

File: README.md [hunk 2]
```diff
@@ -140,6 +140,9 @@ Note: `useTranslation` provides namespaces to the component that you use it in.
 
 ### 5. Advanced configuration
 
+
+#### Passing other config options
+
 If you need to modify more advanced configuration options, you can add a `next-i18next.config.js` file to the root of your project. That file should have a default export. For example:
```
Comment (LeonardDrs) [inline README.md:146]: Hm don't we already tell the user that he needs to create a `next-i18next.config.js` file to import in `next.config.js` for the `i18n` property?

File: src/appWithTranslation.client.test.tsx [hunk 3]
```diff
@@ -48,6 +49,45 @@ describe('appWithTranslation', () => {
     expect(screen.getByText('Hello world')).toBeTruthy()
   })
 
+  it('respects configOverride', () => {
+    const DummyAppConfigOverride = appWithTranslation(() => (
+      <div>Hello world</div>
+    ), {
+      configOverride: 'custom-value',
```
Comment (LeonardDrs) [inline src/appWithTranslation.client.test.tsx:56]: Bikeshed:
May be confusing to use `configOverride` object property in the parameter already named `configOverride`  

PS: I tend to bikeshed a lot, feel free to ignore this kind of comment

**Review by LeonardDrs — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (LeonardDrs) [reply README.md:146]: [Here](https://github.com/isaachinman/next-i18next/pull/972/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R49)

File: README.md (hunk 1 above)
Comment (isaachinman) [reply README.md]: Good point – addressed.

File: README.md (hunk 2 above)
Comment (isaachinman) [reply README.md]: Yep, good catch as well.

File: src/appWithTranslation.client.test.tsx (hunk 3 above)
Comment (isaachinman) [reply src/appWithTranslation.client.test.tsx:56]: I think it's fine in tests, and actually describes what the argument is. If it confuses a contributor in the future, we can modify the naming.

**Review by filipesmedeiros — COMMENTED** [review]
LGTM! I like this solution. Good defaults with nice escape hatch for complex use cases (I know it was already merged, but wanted to chip in ahah)

**Comment by FDiskas** [comment]
Somebody ping me when it will be released?

**Comment by LeonardDrs** [comment]
@FDiskas It's released in the v8.0.0 published 6 days ago.

**Comment by FDiskas** [comment]
thanks - still getting 
```
Error: Error serializing `._nextI18Next.userConfig.use[0]` returned from `getStaticProps`
```
what i'm missing?

**Comment by filipesmedeiros** [comment]
https://github.com/isaachinman/next-i18next/blob/2640dc4fc1234b286e7b690bcb338c23329c0438/README.md#unserialisable-configs

**Pull request merged**